docs/tutorials/integrations/nginx.md

```diff
@@ -0,0 +1,218 @@
+* Since the traffic to and from Conjur is scrambled using secure encryption,
+  passive listeners on your network can't learn anything about the contents of
+  your secrets
+
+For these reasons, it is crucial to set up TLS correctly when you deploy Conjur.
+
+### Protect Your Secrets
+
+Do not leave this to chance: even a small flaw can totally compromise your
+secrets. Setting up Conjur and TLS without the appropriate expertise is like
+packing your own parachute and jumping out of a plane.
+
+That doesn't mean you should close the tab and walk away. It means you should get
+in touch with us and your own security team so we can ensure that you can employ
+Conjur successfully.
+
+[authn]: https://www.conjur.org/api.html#authentication-authenticate-post
+[secret-get]: https://www.conjur.org/api.html#secrets-retrieve-a-secret-get
+
+## Using NGINX To Proxy Traffic With TLS To Conjur
+
+[NGINX][NGINX] is a web proxy that's Free Software and easy to configure. This
+tutorial will show you how to use Docker to install Conjur and NGINX and
+configure them to provide Conjur with TLS.
+
+### Prerequisites
+
+This tutorial requires Docker and a terminal application. Prepare by following
+the prerequisite instructions found on [Install Conjur][prerequisites].
+
+Additionally, you will need the tutorial files from the Conjur source code
+repository. Here's how you get them:
+
+1. Install Git: https://git-scm.com/book/en/v2/Getting-Started-Installing-Git
+1. Clone the Conjur repository
+   
+   In your terminal application, run:
+   ```sh-session
+   $ git clone git@github.com:cyberark/conjur.git
+   ```
+   This will create a folder called `conjur` in the current directory.
+
+### The Good Part
+
+To start out our experience on a high note, let's get the full Conjur+TLS stack
+up and running so we can inspect it.
+
+The tutorial script will install Conjur and NGINX, configure them to work
+together, and connect a Conjur client to the running server. This is a full
+end-to-end working installation to allow you to see how the pieces fit.
+
+```sh-session
+$ # start the Conjur+NGINX tutorial servers
+$ cd conjur/docs/tutorials/integrations/nginx
+$ ./start.sh
+```
+
+Take a look at the logs for the Conjur or NGINX servers:
+
+```sh-session
+$ # in the 'nginx' directory we navigated to previously:
+$ docker-compose logs conjur
+$ docker-compose logs proxy
+```
+
+Run some commands in the Conjur client:
+
+```sh-session
+$ # in the 'nginx' directory we navigated to previously:
+$ docker-compose exec client bash
+# conjur authn whoami
+# conjur list
+```
+
+## Breaking Down the Tutorial
+
+These files show how the proxy setup works:
+
+### docker-compose.yml
+
+This file declares services to be used in the tutorial. Let's break down each declaration:
+
+```yaml
+database:
+  image: postgres:9.3
+```
+
+Conjur requires a Postgres database to store encrypted secrets and other data.
+This service uses the [official Postgres image][postgres-image] from DockerHub.
```

Comment:
Since we're doing best practices here, we should also show how to configure postgres in a way that prevents outside access. This can be accomplished by using TLS authentication, or, in this scenario, more simply by setting up the network so that only the conjur container can connect. You note below that the containers can only connect over private Docker network; even if that's true (not sure if that network isn't shared between all the containers on the host), there's still the matter of nginx container being able to connect to postgres, which is potentially dangerous (if you own nginx you can backdoor into pg).

Reply:
This is interesting @dividedmind I will look into it. To start me out: what if anything do we do for hardening PG within the Conjur Enterprise appliance? Is it doing TLS in there, or does it have a special sub-network or something? Like you say this is a good opportunity to set the standard for the most secure practice we can recommend.

Reply:
Great point @dividedmind.  Why don't we add this to the documentation queue.  I feel like Postgres connection security is beyond the scope of this story.

Reply:
PG in the EE runs locally and uses Linux 'unix' authentication for local user over the domain socket. TCP connections are only allowed for replication (& for audit push) and they require TLS client authentication.